health-monitors/gpu-health-monitor/gpu_health_monitor/tests/test_platform_connector/test_platform_connector.py

```
@@ -32,6 +33,38 @@
+                "device_name": "NVIDIA A100",
+                "nvlinks": [],
+            }
+        ],
+        "nvswitches": [],
+    }
+
+
+@pytest.fixture
+def metadata_file(sample_metadata):
+    """Create a temporary metadata file for testing."""
+    with tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".json") as f:
```

Ruff (0.14.13)

Undefined name `tempfile`

(F821)